src/typing/typeloadFields.ml

```diff
@@ -1032,6 +1032,77 @@ let check_abstract (ctx,cctx,fctx) c cf fd t ret p =
 		| _ ->
 			()
 
+let inherit_doc ctx c cf args p =
+	let inherited = ref None in
+	let doc =
+		match cf.cf_doc with
+		| Some d -> { doc_own = d.doc_own; doc_inherited = d.doc_inherited; }
+		| None -> { doc_own = None; doc_inherited = []; }
+	in
+	cf.cf_doc <- Some doc;
+	(match args with
+	| [] ->
+		delay_late ctx PConnectField (fun() ->
+			match c.cl_super with
+			| None -> ()
+			| Some (csup,_) ->
+				try
+					let cfsup = PMap.find cf.cf_name csup.cl_fields in
+					inherited := gen_doc_text_opt cfsup.cf_doc
+				with Not_found -> ()
+		)
+	| [path] ->
+		let pack, name, sub, field =
+			try
+				match string_list_of_expr_path_raise path with
+				| field :: name :: pack_rev ->
+					(match pack_rev with
+					| module_name :: pack_rev when StringHelper.starts_uppercase_identifier module_name ->
+						List.rev pack_rev, module_name, Some name, field
+					| _ ->
+						List.rev pack_rev, name, None, field
+					)
+				| _ -> raise Exit
+			with Exit | Invalid_argument _ ->
+				error "Invalid argument for @:inheritDoc" p
+		in
+		let t =
+			let tp = { tpackage = pack; tname = name; tsub = sub; tparams = []; } in
+			try Some (Typeload.load_instance ctx (tp,p) true)
+			with Error (Module_not_found _,p2) when p == p2 -> None
+		in
+		flush_pass ctx PBuildClass "inheritDoc";
```

Comment:
Why do we have to do this here?

Reply:
Because we need class structure to be ready at this point to be able to retreive docs of class fields.

Reply:
Oh, right. We don't have to, because actual doc retrieval is delayd to `PConnectField` anyway.